pkg/dockerfile/fast_generator.go

```diff
@@ -227,12 +228,12 @@ func (g *FastGenerator) generateMonobase(lines []string, tmpDir string) ([]strin
 
 	// The only input to monobase.build are these ENV vars
 	// Write them in tmp mount for layer caching
-	err := os.WriteFile(path.Join(tmpDir, "env.txt"), []byte(strings.Join(envs, "\n")), 0o644)
+	err := files.WriteIfDifferent(path.Join(tmpDir, "env.txt"), strings.Join(envs, "\n"))
```

Comment:
Overwriting files with same content changes its timestamp and invalidates layer cache. So only write if content changed.